providers/openshift/authentication.go

```diff
@@ -0,0 +1,217 @@
+package openshift
+
+// this file is partially copied from k8s.io/apiserver/pkg/server/options
```

Comment:
Dropped ApplyTo to prune the dependency on apiserver/pkg/server which brought in about 20 more megs of binary gorp.